src/hash_map.c

```diff
@@ -255,6 +255,13 @@ rcutils_hash_map_init(
     return RCUTILS_RET_INVALID_ARGUMENT;
   }
 
+  // Due to an optimization we use during lookup, we can currently only handle power-of-two
+  // capacities.  Enforce that here.
```

Comment:
Since this is hard requirement, we should add this to doc section in header. (Example, does not even work anymore.)
besides, I think it would be nice to add this failure case in the test.

Reply:
Both very good points.  I'll fix those up.

Reply:
Fixed in b23ad53